lib/mojito.ex

```diff
@@ -278,14 +278,12 @@ defmodule Mojito do
   @spec request(request | request_kwlist) :: {:ok, response} | {:error, error}
   def request(request) do
     with {:ok, valid_request} <- Mojito.Request.validate_request(request) do
-      request_fn =
-        case Keyword.get(valid_request.opts, :pool, true) do
-          true -> fn -> Mojito.Pool.request(valid_request) end
-          false -> fn -> Mojito.Request.Single.request(valid_request) end
-          pool -> fn -> Mojito.Pool.Single.request(pool, valid_request) end
-        end
-
-      request_fn.()
+      case Keyword.get(valid_request.opts, :pool, true) do
+        true -> Mojito.Pool.Poolboy.request(valid_request)
+        false -> Mojito.Request.Single.request(valid_request)
+        pid when is_pid(pid) -> Mojito.Pool.Poolboy.Single.request(pid, valid_request)
+        impl when is_atom(impl) -> impl.request(valid_request)
```

Comment:
There is potential for this to cause an error, in the case that the end user has created a single Poolboy pool under a persistent name, and passed that name to request/1. Passing a pid in is deprecated, so I don't mind.